src/hooks/__internal__/useStableCallback/useStableCallback.ts

```diff
@@ -0,0 +1,16 @@
+import { useRef, useCallback } from "react";
+
+// This is a hook that returns a memoized callback that is guaranteed to be the same reference between renders.
+// This is useful when a stable reference is required to prevent unnecessary re-renders.
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function useStableCallback(callback?: (...args: any[]) => any) {
```

Comment:
```suggestion
export function useStableCallback(callback?: (...args: unknown[]) => unknown) {
```
What do you think of the above so we don't have to disable eslint?